### PR TITLE
fix: Ensure tailmatch is the default for `StaticRoute.directory`

### PR DESCRIPTION
**Pull request by nielsenko**
Currently `router.group('/**').add('/**/', o)` isn't illegal. However, it makes no sense, and `o` will not be looked up on say `a/b` as you might expect. _The tail is a tail_, you don't get to attach anything to a tail.

We managed to trip us self on this, as `addRoute`  is implemented as `group(path).inject(handlerObject)`.  

To fix this Relic has been updated to ^0.14.0 and will now throw whenever an attempt is made on attaching a router on a tail-path. _**UNLESS**_ the attached router contains a single root value, _**AND**_ we are allowed to consume the nodes of the attached router (as opposed to sharing them). The permission to consume is given with the new `attach(..., consume: true)` flag.

Now `addRoute` uses `injectAt`, that is now implemented like:

```dart
  void injectAt(final String path, final InjectableIn<Router<T>> injectable) {
    final subRouter = Router<T>()..inject(injectable);
    attach(path, subRouter, consume: true);
  }
```
Notice it no longer uses `group`, but instead uses `attach` directly with the new `consume: true` flag. 

This will allow the somewhat common corner-case: `webServer.addRoute(MyRoute(), '/**')`, iff `MyRouter` doesn't touch `path` and `injectIn`, ie. it simply attach a single handler at the '/' of the given `router`.
 
```dart
class MyRoute {
  @override
  void injectIn(RelicRouter router) => router.anyOf(methods, '/', _myHandler);
  // ...
}
```

which is the default behaviour inherited from `Route`.

This PR also moves responsibility of the tail-match segment down into `StaticRoute` and ensures no _earlier_ segment uses `**`. This ensures it goes as late as possible, and is never omitted.

Tests has been updated not to use `/**` when setting up `FlutterRoute`, `SpaRoute`, and `StaticRoute.directory` (this is now left to `StatidRoute.directory` ctor). Tests for the `/` corner-case has been added.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None. No interface changes. Semantics has changed, but this is reasonably a bug-fix.

### CODE REVIEW

**Review by exaby73 — APPROVED** [review]
LGTM

**Comment by nielsenko** [comment]
@coderabbitai review

File: packages/serverpod/test/web_server/add_route_test.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+void main() {
+  final portZeroConfig = ServerConfig(
+    port: 0,
+    publicScheme: 'http',
+    publicHost: 'localhost',
+    publicPort: 0,
+  );
+
+  group('WebServer.addRoute', () {
```
Comment (marcelomendoncasoares) [inline packages/serverpod/test/web_server/add_route_test.dart]: Missing "Given" on the test description.

File: packages/serverpod/test/web_server/add_route_test.dart [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+              ),
+              '/**',
+            ),
+            throwsArgumentError,
+          );
+        },
+      );
+
+      test(
+        'rejects route with non-rooted path',
```
Comment (marcelomendoncasoares) [inline packages/serverpod/test/web_server/add_route_test.dart]: Missing "then" to compose the Given/When/Then pattern. Same for below tests and groups.

Comment (nielsenko) [reply packages/serverpod/test/web_server/add_route_test.dart]: Seems to be a common theme 😬

File: packages/serverpod/test/web_server/add_route_test.dart (hunk 1 above)
Comment (nielsenko) [reply packages/serverpod/test/web_server/add_route_test.dart]: Fixed in b49fc07

File: packages/serverpod/test/web_server/add_route_test.dart (hunk 2 above)
Comment (nielsenko) [reply packages/serverpod/test/web_server/add_route_test.dart]: Fixed in b49fc07

**Comment by SandPod** [comment]
@coderabbitai please create an issue for adding test that validate the routing behavior of `StaticRoute.file`. It seems like this is never tested.
